### DISCUSSION TITLE
Nested Layouts

### DISCUSSION THREAD

**Discussion by MohamedFadelMoumeni** [vercel/next.js#43821] [Help]
Hello guys,

I have a question about nested layouts  ,I'm building an ecommerce app where the path of listing all the products of  a single category is` www.example.com/products/category_name`

<img width="960" alt="tty" src="https://user-images.githubusercontent.com/56879358/206259160-8321b546-f598-4e3d-8a05-67f0ca298a6c.png">

The red part is the layout and the yellow part is changing depends on the category.

But I want when someone visit ` www.example.com/products/category_name/product_id` (this path for a signle produt) display a new layout .

This is the folder structure of this feature.
<img width="233" alt="folder" src="https://user-images.githubusercontent.com/56879358/206259995-d35c4d0b-da93-4522-87f3-e5638c9e4506.png">

**Comment by cjdunteman** [comment]
Inside of categorie you can have another directory, [product], with a page.js file. See [here](https://beta.nextjs.org/docs/api-reference/file-conventions/page#params-optional)

**Reply by MohamedFadelMoumeni** [reply]
Yes I know but the [product] will inheritance the products layout I don't want that , I want to create a new layout for [product] directory

**Reply by cjdunteman** [reply]
You can create a `layout.tsx` inside of [product] to give a different layout for all the product pages. See [Nested Layouts](https://beta.nextjs.org/docs/routing/pages-and-layouts#nesting-layouts).

**Reply by cjdunteman** [reply]
Depending on what exactly you want to do you may need a [Template](https://beta.nextjs.org/docs/routing/pages-and-layouts#templates).

**Reply by MohamedFadelMoumeni** [reply]
If I create a new `layout.tsx` for [product] it will be combined with the  products layout . That's not what I want

**Reply by MohamedFadelMoumeni** [reply]
Let me try to explain again the path `/products/football` wil display all the product cards related to football The `products` directory has a `layout.tsx` the layout contains the filter bar on the left and searchbox on the top  as you can see in the image .

<img width="960" alt="206259160-8321b546-f598-4e3d-8a05-67f0ca298a6c" src="https://user-images.githubusercontent.com/56879358/206577401-22fdc50d-4805-448d-adeb-1fcd26584554.png">

So when you navigate to another category only the product cards got changed .

But when someone clicks on the product card , will be redirected to `/products/category/product_name`.
 **The problem is that the Products layout that contains the searchbox and the filter bar will be visible  in Product Page I want it to be invisible on the Product page**

**Reply by cjdunteman** [reply]
You should put the searchbox and filter bar inside `/products/category/page.tsx` instead of `/products/category/layout.tsx`. 

Layout files are for shared UI, each layout wraps the layout below it. So `/app/layout.tsx` wraps `/app/products/category/layout.tsx` wraps `/app/products/category/products/layout.tsx`.

Only put UI in a layout.tsx file if you want it included in pages beneath it.

**Comment by MohamedFadelMoumeni** [comment]
@cjdunteman Yes I know but the [product] will inheritance the the products layout I don't want that , I want to create a new layout for [product] directory
